src/adc.rs

```diff
@@ -502,23 +484,65 @@ adc_hal! {
     feature = "stm32f103",
 ))]
 adc_hal! {
-    ADC1: (adc1),
     ADC2: (adc2),
 }
 
-pub struct AdcPayload<PIN: Channel<ADC1>> {
+pub struct AdcPayload<PINS> {
     adc: Adc<ADC1>,
-    pin: PIN,
+    pins: PINS,
 }
 
-pub type AdcDma<PIN> = RxDma<AdcPayload<PIN>, C1>;
+pub trait ChannelTimeSequence {
+    /// Set ADC sampling time for particular channel
+    fn set_channel_sample_time(&mut self, chan: u8, sample_time: SampleTime);
+    /// ADC Set a Regular Channel Conversion Sequence
+    ///
+    /// Define a sequence of channels to be converted as a regular group.
+    fn set_regular_sequence (&mut self, channels: &[u8]);
+}
 
-impl<PIN> Receive for AdcDma<PIN> where PIN: Channel<ADC1> {
+/// Set channel sequence and sample times for custom pins
+///
+/// Example:
+/// ```rust, ignore
+/// pub struct AdcPins(PA0<Analog>, PA2<Analog>);
+/// impl SetChannels<AdcPins> for Adc<ADC1> {
+///     fn set_samples(&mut self) {
+///         self.set_channel_sample_time(0, adc::SampleTime::T_28);
+///         self.set_channel_sample_time(2, adc::SampleTime::T_28);
+///     }
+///     fn set_sequence(&mut self) {
+///         self.set_regular_sequence(&[0, 2, 0, 2]);
+///     }
+/// }
+/// ```
+pub trait SetChannels<PINS>: ChannelTimeSequence {
+    fn set_samples(&mut self);
+    fn set_sequence(&mut self);
+}
+
+impl<PIN> SetChannels<PIN> for Adc<ADC1>
```

Comment:
In basic ```OneShot``` implementation single channel sequence is configured in ```convert``` function. In the case of multiple channels, we have to implement ```SetChannel``` anyway to specify particular list of channels. So could you please clarify why do we need this implementation of SetChannels for single channel ? Does it help to maintain backward compatibility for ```with_dma``` ?